### PR TITLE
Make sure we fail on invalid em:type for dictionaries, fixes #364

### LINKED ISSUE #364: Prevent dictionaries that do not have the em:type set correctly
As per mozilla/olympia#1093

### PR DESCRIPTION
**Pull request by EnTeQuAk**

### CODE REVIEW

**Comment by EnTeQuAk** [comment]
@wagnerand Can you review this PR, specifically if the wording if fitting?

File: validator/typedetection.py [hunk 1]
```diff
@@ -63,6 +63,15 @@ def detect_type(err, install_rdf=None, xpi_package=None):
...
                file_.startswith('dictionaries/')):
+        em_type = err.get_resource('em:type')
+
+        if em_type != 64:
+            err.error(('typedetection',
+                       'dictionary_valid_type',
+                       'invalid_em_type'),
+                      'Invalid <em:type> value.',
+                      'The package appears to be a dictionary but does not have '
+                      'the correct <em:type> set.')
```
Comment (wagnerand) [inline validator/typedetection.py]: I'd do:
`'the correct <em:type> set in the install manifest.'`

**Comment by wagnerand** [comment]
r+wc

File: validator/typedetection.py [hunk 2]
```diff
@@ -63,6 +63,15 @@ def detect_type(err, install_rdf=None, xpi_package=None):
 
     if any(file_ for file_ in xpi_package if
                file_.startswith('dictionaries/')):
+        em_type = err.get_resource('em:type')
```
Comment (wagnerand) [inline validator/typedetection.py]: Any reason why you don't use `type_` here?

Comment (EnTeQuAk) [inline validator/typedetection.py]: fixed in master

**Pull request merged**